app/Dockerfile

```diff
@@ -2,26 +2,47 @@
 # https://hub.docker.com/_/python
 
 # The build stage that will be used to deploy to the various environments
-# needs to be called `release` in order to integrate with the repo's
-# top-level Makefile
-FROM python:3-slim AS release
-
-# Install poetry, the package manager.
-# https://python-poetry.org
-RUN pip install poetry
+# This is what the other stages will call as their base
+FROM python:3.11-slim-bullseye AS build
 
+# Keep container packages up-to-date.
 RUN apt-get update \
     && apt-get install --no-install-recommends --yes \
     build-essential \
     libpq-dev \
     postgresql
 
+# Install poetry, the package manager.
+# https://python-poetry.org
+RUN pip install --no-cache-dir poetry==1.3.1
```

Comment:
Added a blurb about it, but I'll throw it here too
```
# Install poetry, the package manager - https://python-poetry.org
# We use --no-cache-dir here because the pip caches normally, and if 
#  we don't cache, it will reduce the overall image size
# The version is pinned so that there is a known good configuration 
#  that we have better control over vs a range that can be used
```